### PR TITLE
Instrument newer versions

### PR DESCRIPTION
**Pull request by Symbianx**
# Description

The latest confluent-kafka version is 1.9.2, which is not being instrumented due to the current restricted version.

This change increases the confluent-kafka version check to also instrument versions greater or equal to 1.9.0.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Ran the existing unit tests with the changes and it passed. To make sure the
tests were running with the latest confluent-kafka version also hardcoded the
`1.9.2` version in the requirements and ran the test suite again and the test
passed.

# Does This PR Require a Core Repo Change?

- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Documentation has been updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#1409](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1409))
 - Strip leading comments from SQL queries when generating the span name.
   ([#1434](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1434))
+- `opentelemetry-instrumentation-confluent-kafka` Add support for the latest versions of the library.
```
Comment (srikanthccv) [inline CHANGELOG.md:31]: Add link to PR

**Review by srikanthccv — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Symbianx) [reply CHANGELOG.md:31]: Link added 👍

**Comment by srikanthccv** [comment]
You can skip the tests on 3.11 and add a FIXME note.

**Comment by Symbianx** [comment]
@srikanthccv I think the last change was enough to disable the tests for 3.11 and pypy ([which is not supported and never will be](https://github.com/confluentinc/confluent-kafka-python/issues/298)).

**Pull request merged**